### DISCUSSION TITLE
Multi Vendor Shop

### DISCUSSION THREAD

**Discussion by xKeRnx** [givanz/Vvveb#58] [Ideas]
Would it be possible to create a plugin to change the store into a multi vendor store, everyone can open his own store and sell items...on the main shop page there is then the category or the selection of stores and can also come via a direct link to a specific store.
The website operator can define a separate commission for each store operator or create a standard commission per sale.

**Comment by givanz** [comment]
I expanded vendor support in admin by adding a new vendor admin role that has access only to orders and products and an associated admin user field for the vendor.

I also added capabilities to roles to restrict editing only self product and posts for admin users.

To list all vendors you can use the vendors component either by adding it from the page builder or adding the code directly into the page.

```html
<ul class="list-unstyled" data-v-component-product-vendors>
	<li data-v-vendor>
		  <a href="/manufacturer/mango" data-v-vendor-url><span data-v-vendor-name>Mango</span></a>
	</li>
	<li data-v-vendor>
		  <a href="/manufacturer/mango" data-v-vendor-url><span data-v-vendor-name>Assos</span></a>
	</li>
	<li data-v-vendor>
		  <a href="/manufacturer/mango" data-v-vendor-url><span data-v-vendor-name>Brand name</span></a>
	</li>
</ul>
```

For vendor signup and other advanced features I made a new *multi-vendor* plugin that adds a signup page `/vendor-signup` that creates a new vendor admin account.

The link can be added to menu or in any page.

Commissions and other features will be added gradually in further updates.

The plugin is bundled in *latest.zip* and will be moved to https://plugins.vvveb.com after it will be more complete with more features.

**Comment by BastianHammer** [comment]
Hi @givanz 

that looks highly interesting.
How would the other required features work like:

- payment
- vendor based shipping cost
- the vendor dashboard
- vendor / customer communication
- etc.

**Reply by givanz** [reply]
Hi

* Payment for vendors will be provided through plugins, the first one to be implemented is Stripe Connect.

* This feature has low priority, at first the multi vendor plugin will be used for digital products later it will include vendor specific shipping rules that will be configured from vendor settings page.

* The vendor does not have a separate dashboard, vendor users are admin users with vendor role https://demo.vvveb.com/admin/index.php?module=admin/role&role_id=8 that allows access only to vendor orders and products and later to vendor settings page. 

* By communication you mean some chat system? The vendor page will include optional vendor email and phone number details to allow communication.

**Reply by BastianHammer** [reply]
Thank you for the answers :)
